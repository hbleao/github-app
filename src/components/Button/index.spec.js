import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import user from '@testing-library/user-event';

import { Button } from '.';

const makeSut = ({
  onClick = () => null,
  styles = 'is-green',
  children = 'click here',
}) => {
  render(
    <Button onClick={onClick} styles={styles}>
      {children}
    </Button>
  );
};

describe('Button', () => {
  it('should be able render correctly', () => {
    makeSut({});

    const buttonTextDefault = screen.getByText(/click here/gi);
    expect(buttonTextDefault).toBeTruthy();
  });

  it('should be able to render component with text confirm', () => {
    makeSut({ children: 'confirm' });

    const buttonTextDefault = screen.getByText(/confirm/gi);
    expect(buttonTextDefault).toBeTruthy();
  });

  it('should be able to click into button', () => {
    const handleClick = jest.fn();
    makeSut({ onClick: handleClick });

    const buttonTextDefault = screen.getByText(/click here/gi);
    user.click(buttonTextDefault);
    user.click(buttonTextDefault);

    expect(handleClick).toBeCalledTimes(2);
  });

  it('should be able to render a component Button with styles default props', () => {
    makeSut({});

    const buttonWithStyleDefault = document.querySelector('.is-green');
    expect(buttonWithStyleDefault).toBeTruthy();
  });

  it('should be able to render snapshot', () => {
    const handleClick = jest.fn();
    const component = renderer.create(
      <Button onClick={handleClick}>Click!</Button>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
