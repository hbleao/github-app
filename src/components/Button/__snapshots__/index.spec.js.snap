// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be able to render snapshot 1`] = `
<button
  className="button is-green"
  onClick={[MockFunction]}
  type="button"
>
  Click!
</button>
`;
